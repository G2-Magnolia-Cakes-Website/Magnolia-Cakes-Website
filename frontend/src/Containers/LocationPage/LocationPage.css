.location-page-wrapper {
  display: flex;
  justify-content: space-between;
  margin: 1.5rem;
  flex: 1;
}

.location-page-wrapper .location-info {
  justify-content: center;
  align-items: center;
  margin: auto 5rem auto 5rem;
  height: inherit;
  padding-top: 2rem;
  padding-bottom: 2rem;
}

.location-page-wrapper .location-info h2 {
  font-size: 64px;
  /* font-family: "Pacifico", cursive; */
  font-family: "Dancing Script", cursive;
  color: #daa41a;
  /* color: black; */
}

.location-page-wrapper .location-info p {
  font-size: 24px;
  color: black;
  margin: 1rem 2rem 0 2rem;
}

.location-page-wrapper .location-info p:not(:last-child) {
  margin-bottom: 4rem;
}

@media (max-width: 768px) and (min-width: 480px) {
  .location-page-wrapper {
    flex-direction: column;
    margin: 2rem;
  }

  .location-page-wrapper .location-info {
    margin: 0 0 3rem 0;
  }
}

@media (max-width: 480px) {
  .location-page-wrapper {
    flex-direction: column;
    margin: 0.5rem;
  }

  .location-page-wrapper .location-info {
    margin: 2rem;
  }

  .location-page-wrapper .location-info h2 {
    font-size: 40px;
  }

  .location-page-wrapper .location-info p {
    font-size: 18px;
    margin: 1rem 0.5rem 0 0.5rem;
  }

  .location-page-wrapper .location-info p:not(:last-child) {
    margin-bottom: 2rem;
  }
}
