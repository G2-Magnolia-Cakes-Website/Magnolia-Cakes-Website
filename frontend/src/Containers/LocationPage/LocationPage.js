import React, { useEffect, useState } from "react";
import ShopMap from "Components/ShopMap/ShopMap";
import { parseStringToParagraphsByNewline } from "utils/parseParagraphs";
import BarLoader from "react-spinners/BarLoader";

import "./LocationPage.css";

const LocationPage = ({ api }) => {
  const [locationContent, setLocationContent] = useState({
    business_hours_heading: "Business Hours",
    business_hours_info: "Coming Soon...",
    location_heading: "Where to Find Us?",
    location_info: "Coming Soon...",
  });

  // Loading
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    setLoading(true);
    // Make a GET request using the passed api instance
    api
      .get("/api/location-page/")
      .then((response) => {
        // Set the retrieved content in the state
        setLocationContent(response.data);
        setLoading(false);
      })
      .catch((error) => {
        console.error("Error fetching data:", error);
      });
  }, [api]);

  return (
    <>
      <div className="location-page-wrapper">
        <div className="location-info">
          <h2>{locationContent.location_heading}</h2>
          {parseStringToParagraphsByNewline(locationContent.location_info).map(
            (paragraph) => (
              <p className="info">{paragraph}</p>
            )
          )}
          <div className="spacer" />
          <h2>{locationContent.business_hours_heading}</h2>
          {parseStringToParagraphsByNewline(
            locationContent.business_hours_info
          ).map((paragraph) => (
            <p className="info">{paragraph}</p>
          ))}
        </div>
        <ShopMap />
      </div>
      <BarLoader
        loading={loading}
        aria-label="Loading Spinner"
        data-testid="loader"
        width={"100%"}
      />
    </>
  );
};

export default LocationPage;
