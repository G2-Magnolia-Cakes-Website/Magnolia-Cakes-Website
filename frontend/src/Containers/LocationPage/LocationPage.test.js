import React from "react";
import renderer from "react-test-renderer";
import LocationPage from "./LocationPage";

describe("test LocationPage", () => {
  it("renders LocationPage", () => {
    const wrapper = renderer.create(<LocationPage />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
