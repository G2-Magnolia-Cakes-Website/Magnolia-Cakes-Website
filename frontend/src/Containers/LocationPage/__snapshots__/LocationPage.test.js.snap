// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LocationPage renders LocationPage 1`] = `
<div
  className="location-page-wrapper"
>
  <div
    className="location-info"
  >
    <h2>
      Where to find us?
    </h2>
    <p>
      Melbourne
      <br />
      VIC 3752
    </p>
    <h2>
      Business Hours
    </h2>
    <p>
      Monday - Sunday
      <br />
      9:00 am - 18:00 pm
    </p>
  </div>
</div>
`;
